api/admin/controller/__init__.py

```diff
@@ -1112,6 +1125,8 @@ def _create_or_update_list(
             list, is_new = create(
                 self._db, CustomList, name=name, data_source=data_source
             )
+            if list is None:
+                return INTERNAL_SERVER_ERROR.detailed("Could not create the list.")
```

Comment:
I've used internal server error in several cases where the "create" would have failed because that is essentially what it is, the server failed to write to the DB for some reason.

Reply:
I some cases to avoid changing the function definition I've simply raised ValueErrors, which would translate to an INTERNAL_SERVER_ERROR higher up in the stack.